### ISSUE TITLE
redux-observable getInitialProps example

### ISSUE THREAD

**Issue by oste** [zeit/next.js#8817]
# Feature request

## Is your feature request related to a problem? Please describe.

I found the [redux-observable](https://github.com/zeit/next.js/tree/canary/examples/with-redux-observable) example really useful. The awkward part, as outlined in the README is how actions are dispatched in `getInitialProps` shown here https://github.com/zeit/next.js/blob/canary/examples/with-redux-observable/pages/index.js#L12-L17

## Describe the solution you'd like

Was just wondering if this is still the best way forward. I assume we are still waiting on a good solution for SSR in redux-observable itself. If this is still the best bet I was trying to brainstorm an abstraction for it to keep things DRY when doing this more than once.

**Comment by timneutkens** [comment]
`getInitialProps` is needed as React SSR needs the full state to render the app to HTML.

**Comment by oste** [comment]
@timneutkens I agree. I was more looking for an abstraction to the code shown on lines 12-17. I just don't like copying that boilerplate to every component. Ideally redux observable could support SSR and have something built-in but that progress doesn't look good. I did see this comment but the solution didn't seem to get much traction back then. https://github.com/redux-observable/redux-observable/issues/47#issuecomment-341393418

Was just wondering if anyone wanted to brainstorm on this. Maybe it can lead to a proposal for redux-observable srr support

**Comment by oste** [comment]
@timneutkens I agree. I was looking for ideas to

**Comment by Timer** [comment]
Honestly, this seems pretty idiomatic! If you have a better suggestion, please feel free to send a PR -- but I haven't seen any 👍s on this issue. 

Thanks!

**Comment by oste** [comment]
@timer there has been a lot of recent good discussion on the related redux-observable issue. Will wait to see what decisions are made there before updating the example

**Comment by Timer** [comment]
Great, thanks!

**Comment by fulopdaniel** [comment]
Hey, I'd just like to point out that if you use the `delay` operator in one of your epics the `await rootEpic(...)` promise will never resolve. Use `timer` instead if possible.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
